lib/flex_algo.c

```diff
@@ -0,0 +1,307 @@
+/*
+ * IGP Flexible Algorithm Library module.
+ * Copyright (C) 2022  Vmware, Inc.
+ *		       Pushpasis Sarkar <spushpasis@vmware.com>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; see the file COPYING; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include <zebra.h>
+#include "flex_algo.h"
+
+/* Data-structure definitions */
+
+struct flex_algo_db {
+	struct flex_algo_defn *defns[FLEX_ALGO_ID_MAX+1];	
+};
+
+/* Extern variables. */
+
+
+/* Forward Prototypes. */
+
+/* Private function definitions */
+
+/* Public function definitions */
+
+struct flex_algo_db *flex_algo_db_init(void)
+{
+	struct flex_algo_db *db;
+
+	db = calloc(1, sizeof(struct flex_algo_db));
```

Comment:
No.  Please actually use XCALLOC/XMALLOC and XFREE.  This is a non-starter from me

Reply:
Ok. Will take care of that.